cmd/nerdctl/system_prune.go

```diff
@@ -99,5 +102,68 @@ func systemPruneAction(cmd *cobra.Command, args []string) error {
 			return err
 		}
 	}
-	return imagePrune(cmd, client, ctx)
+	if err := imagePrune(cmd, client, ctx); err != nil {
+		return nil
+	}
+	prunedObjects, err := buildCachePrune(ctx, cmd, all)
```

Comment:
all should always be true imo.

Reply:
YES! Updated!

Reply:
docker system prune gives the following:
```
➜  ~ docker system prune
WARNING! This will remove:
  - all stopped containers
  - all networks not used by at least one container
  - all dangling images
  - all dangling build cache
```

There is no option to differentiate if we should or we should not prune the entire buildCache. 
The all parameter is used to differentiate between dangling and unused which is not true in this case.
If this makes sense, can we make the 3rd parameter as true.

Reply:
There are some different message from `docker system prune --all`:

```
$ docker system prune -a    
WARNING! This will remove:
  - all stopped containers
  - all networks not used by at least one container
  - all images without at least one container associated to them
  - all build cache

Are you sure you want to continue? [y/N] 
```

And the message from `docker builder prune` also introduce similar flags:

```
  ~ ──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────  system at  10:54:59
❯ docker builder prune       
WARNING! This will remove all dangling build cache. Are you sure you want to continue? [y/N] ^C
  ~ ──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────── ✘ INT  system at  10:55:05
❯ docker builder prune -a    
WARNING! This will remove all build cache. Are you sure you want to continue? [y/N] ^C
  ~ ──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────── ✘ INT  system at  10:55:08
❯ docker builder prune --help

Usage:  docker builder prune

Remove build cache

Options:
  -a, --all                  Remove all unused build cache, not just dangling ones
      --filter filter        Provide filter values (e.g. 'until=24h')
  -f, --force                Do not prompt for confirmation
      --keep-storage bytes   Amount of disk space to keep for cache

```

Reply:
so with `nerdctl system prune -a`, I think we should use `true` as the `buildCachePrune()`'s third parameter.